card/mod/05_standard/spec/set/all/history_spec.rb

```diff
@@ -68,11 +68,41 @@
         end
         it 'adds delete action' do
           expect(action.action_type).to eq(:delete)
-        end  
-        it 'adds trash change' do  
-          expect(action.changes.last.field).to eq('trash')
-          expect(action.changes.last.value).to be_truthy
         end
+        it 'adds trash change' do
+          expect(action.card_changes.last.field).to eq('trash')
+          expect(action.card_changes.last.value).to be_truthy
+        end
+      end
+
+      context 'when creation fails' do
+        it "doesn't create an act" do
+          Card::Auth.as(:anonymous) do
+            act_count = Card::Act.count
+            Card.create :name=>"create fail"
+            expect(Card::Act.count).to eq act_count
+          end
+        end
+
+        it "doesn't create an action" do
+          Card::Auth.as(:anonymous) do
+            action_count = Card::Action.count
+            Card.create :name=>"create fail"
+            expect(Card::Action.count).to eq action_count
+          end
+        end
```

Comment:
you were right, Ethan, if the transaction fails it takes care of other stuff that was created in the same process